### PR TITLE
feat: estimate region wal size

### PR DESCRIPTION
**Pull request by QuenKar**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- memtable size is more accurate (now include timestamp and op type size)
- use memtable size to estimate wal size.

#### details
1. construct different size of rows to insert into or delete from table, and record relevant memtable size and wal size.
2. compute linear regression of these data, and get a rough slope, and use regression line' slope to estimate wal size.

SCR-20231024-sztu

the `slope` is not very accurate, and I also don't think about the `interception` because wal writes some metadata in `WalEntry`, and if the rows size is small, it's inaccurate.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/2551

### CODE REVIEW

File: src/mito2/src/engine.rs [hunk 1]
```diff
@@ -95,6 +95,11 @@ impl MitoEngine {
     pub(crate) fn get_region(&self, id: RegionId) -> Option<crate::region::MitoRegionRef> {
         self.inner.workers.get_region(id)
     }
+
+    /// Returns the estimated size of the region's WAL.
+    pub fn get_region_wal_size(&self, id: RegionId) -> Result<usize> {
```
Comment (evenyag) [inline src/mito2/src/engine.rs:100]: nitpick: put the pub method before `pub(crate)` and private methods (moving to L88).

Comment (evenyag) [inline src/mito2/src/engine.rs:100]: non-blocking: Well, I think we can return a struct that contains the WAL size, memtable size, manifest size and SST size of the region in the future. Then we don't need this method.

**Review by evenyag — APPROVED** [review]

File: src/mito2/src/engine.rs (hunk 1 above)
Comment (QuenKar) [reply src/mito2/src/engine.rs:100]: ok

Comment (QuenKar) [reply src/mito2/src/engine.rs]: I will remove this method

**Review by waynexia — APPROVED** [review]

**Pull request merged**